### PR TITLE
Add egress and ingress worker filters

### PR DESCRIPTION
**Pull request by irenarindos**

### CODE REVIEW

File: internal/provider/resource_target_test.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ resource "boundary_target" "foo" {
 	depends_on  = [boundary_role.proj1_admin]
 	session_max_seconds = 6000
 	session_connection_limit = 6
-	worker_filter = "type == \"foo\""
```
[the remark this answers is not in the document]
Comment (tmessi) [reply internal/provider/resource_target_test.go:103]: It seems that `worker_filter` is not supported when creating a new target.

File: go.mod [hunk 2]
```diff
@@ -2,10 +2,13 @@ module github.com/hashicorp/terraform-provider-boundary
 
 go 1.19
 
+// TODO REMOVE THIS AFTER THE 0.12.0 RELEASE
+replace github.com/hashicorp/boundary => ../boundary
+
```
Comment (tmessi) [inline go.mod:7]: This should be removed

File: go.mod [hunk 3]
```diff
@@ -2,10 +2,13 @@ module github.com/hashicorp/terraform-provider-boundary
...
 require (
-	github.com/hashicorp/boundary v0.11.1
-	github.com/hashicorp/boundary/api v0.0.32
-	github.com/hashicorp/boundary/sdk v0.0.26
+	github.com/hashicorp/boundary v0.12.0
+	github.com/hashicorp/boundary/api v0.0.33
+	github.com/hashicorp/boundary/sdk v0.0.29
```
Comment (tmessi) [inline go.mod:11]: These should be updated to the latest, which I believe is `v0.0.34` for api and `.v0.0.30` for sdk.

**Review by tmessi — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (irenarindos) [reply go.mod]: Thanks, done!

File: go.mod (hunk 3 above)
Comment (irenarindos) [reply go.mod]: Thanks!

File: go.mod [hunk 4]
```diff
@@ -3,9 +3,9 @@ module github.com/hashicorp/terraform-provider-boundary
 go 1.19
 
 require (
-	github.com/hashicorp/boundary v0.11.1
-	github.com/hashicorp/boundary/api v0.0.32
-	github.com/hashicorp/boundary/sdk v0.0.26
+	github.com/hashicorp/boundary v0.12.0
+	github.com/hashicorp/boundary/api v0.0.33
```
Comment (tmessi) [inline go.mod:7]: This still needs to be `v0.0.34`

**Review by tmessi — APPROVED** [review]

**Pull request merged**